packages/connect/src/core/index.ts

```diff
@@ -546,6 +558,17 @@ const onCall = async (context: CoreContext, message: IFrameCallMessage) => {
             await waitForPopup(context);
             // show message about transport
             sendCoreMessage(createUiMessage(UI.TRANSPORT));
+
+            // Retry onCallDevice again
+            // NOTE: this should change after multi-transports refactor, where transport will be always alive
+            if (transportReconnect && (env === 'web' || env === 'webextension')) {
+                // wait 10s between reconnects
+                const reconnectTime = 10000 - (Date.now() - initDeviceTime);
+                await resolveAfter(reconnectTime).promise;
+
+                // call onCallDevice again recursively
+                return await onCallDevice(context, message, method);
```

Comment:
wouldnt it be better to remain hanging here on some promise, that would resolve after 
 ```
deviceList.on('transport-start')
```
is emitted and only after that it would retry the onCallDevice call?

Reply:
Interesting idea, I will take a look at the emitted events

Reply:
Just pushed the change to this, seems to be working well